ciao-controller/workload.go

```diff
@@ -122,6 +137,24 @@ func (c *controller) CreateWorkload(req types.Workload) (types.Workload, error)
 		return req, err
 	}
 
+	// check to see if this is a new tenant. If so, we need to add
+	// them to the datastore. We do not however want to launch a
+	// CNCI yet since this might be a request to upload a CNCI
+	// workload. Instead, we'll add the new tenant directly to the
+	// datastore. On first launch request, if the tenant doesn't yet
+	// have a cnci, it will get launched for them then.
+	tenant, err := c.ds.GetTenant(req.TenantID)
```

Comment:
so - I thought about this - right now anytime confirmTenant() is used (in many of the volume calls) if there isn't a CNCI workload ID, you'll get an error returned and the call will fail. I personally think this is ok - it means the admin hasn't configured the cluster properly. Even though in theory we could of course make it possible to access volume services without the CNCI being configured properly, I'd rather it cause an error that forces them to finish setting up the cluster. the workloads api is special of course.